internal/fs/async_fetch_test.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fs_test
+
+import (
+	"os"
+	"path"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/googlecloudplatform/gcsfuse/internal/fs"
+	"github.com/googlecloudplatform/gcsfuse/internal/fs/inode"
+	"github.com/googlecloudplatform/gcsfuse/internal/gcsx"
+	"github.com/jacobsa/fuse/fuseops"
+	"github.com/jacobsa/gcloud/gcs"
+	"github.com/jacobsa/gcloud/gcs/gcsfake"
+	. "github.com/jacobsa/ogletest"
+	"github.com/jacobsa/timeutil"
+	"golang.org/x/net/context"
+)
+
+const dirInodeID = 17
+const dirInodeName = "foo/"
+const dirMode os.FileMode = 0712 | os.ModeDir
+const fileUnderDir = "bar"
+const implicitDirName = "baz"
+const typeCacheTTL = time.Second
+const uid = 123
+const gid = 456
+const tmpObjectPrefix = ".gcsfuse_tmp/"
+const appendThreshold = 1
+const fakeBucketName = "some_bucket"
+
+type AsyncFetchTest struct {
```

Comment:
why is the test file called AsyncFetchtest and not dir_handle_test??

Reply:
This was just testing the async function call but makes sense.I will rename it.